plugins/outputs/opensearch/opensearch.go

```diff
@@ -0,0 +1,443 @@
+
+	buffer.WriteString(strconv.FormatInt(m.Time().Local().UnixNano(), 10))
+	buffer.WriteString(m.Name())
+	buffer.WriteString(strconv.FormatUint(m.HashID(), 10))
+
+	return fmt.Sprintf("%x", sha256.Sum256(buffer.Bytes()))
+}
+
+func (o *Opensearch) Write(metrics []telegraf.Metric) error {
+	// get indexers based on unique pipeline values
+	indexers := getTargetIndexers(metrics, o)
+	if len(indexers) == 0 {
+		return fmt.Errorf("failed to instantiate opensearch bulkindexer")
+	}
+	
+	ctx, cancel := context.WithTimeout(context.Background(), time.Duration(5000000000))
+	defer cancel()
+
+	onSucc := func(ctx context.Context, item opensearchutil.BulkIndexerItem, res opensearchutil.BulkIndexerResponseItem) {
+		fmt.Printf("Indexed to Opensearch with status- [%d] Result- %s DocumentID- %s \n", res.Status, res.Result, res.DocumentID)
+	}
+	onFail := func(ctx context.Context, item opensearchutil.BulkIndexerItem, res opensearchutil.BulkIndexerResponseItem, err error) {
+		if err != nil {
+			fmt.Printf("ERROR while Opensearch bulkIndexing: %s \n", err)
+		} else {
+			fmt.Printf("ERROR while Opensearch bulkIndexing: %s: %s \n", res.Error.Type, res.Error.Reason)
+		}
+	}
+
+	for _, metric := range metrics {
+		var name = metric.Name()
+
+		// index name has to be re-evaluated each time for telegraf
+		// to send the metric to the correct time-based index
+		indexName := o.GetIndexName(o.IndexName, metric.Time(), o.tagKeys, metric.Tags())
+
+		// Handle NaN and inf field-values
+		fields := make(map[string]interface{})
+		for k, value := range metric.Fields() {
+			v, ok := value.(float64)
+			if !ok || o.FloatHandling == "none" || !(math.IsNaN(v) || math.IsInf(v, 0)) {
+				fields[k] = value
+				continue
+			}
+			if o.FloatHandling == "drop" {
+				continue
+			}
+
+			if math.IsNaN(v) || math.IsInf(v, 1) {
+				fields[k] = o.FloatReplacement
+			} else {
+				fields[k] = -o.FloatReplacement
+			}
+		}
+
+		m := make(map[string]interface{})
+
+		m["@timestamp"] = metric.Time()
+		m["measurement_name"] = name
+		m["tag"] = metric.Tags()
+		m[name] = fields
+
+		body, err := json.Marshal(m)
+		if err != nil {
+			return fmt.Errorf("failed to marshal body: %w", err)
+		}
+
+		bulkIndxrItem := opensearchutil.BulkIndexerItem{
+			Action: "index",
+			Index: indexName,
+			Body:      strings.NewReader(string(body)),
+			OnSuccess: onSucc,
+			OnFailure: onFail,
+		}
+		if o.ForceDocumentID {
+			bulkIndxrItem.DocumentID = getPointID(metric)
+		}
+
+		if o.UsePipeline != "" {
+			if pipelineName := o.getPipelineName(o.pipelineName, o.pipelineTagKeys, metric.Tags()); pipelineName != "" {
+				if indexers[pipelineName] != nil {
+					indexers[pipelineName].Add(ctx, bulkIndxrItem)
+					continue
+				}
+			}
+		}
+		
+		indexers["default"].Add(ctx, bulkIndxrItem)
+	}
+
+	for _, bulkIndxr := range indexers {
+		if err := bulkIndxr.Close(ctx); err != nil {
+			return fmt.Errorf("error sending bulk request to Opensearch: %s", err)
```

Comment:
Errors need to be wrapped
```suggestion
			return fmt.Errorf("error sending bulk request to Opensearch: %w", err)
```

Reply:
done